pulp_2to3_migration/tests/functional/test_migration_behaviour.py

```diff
@@ -0,0 +1,337 @@
+        pulp2repositories = self.pulp2repositories_api.list(ordering='pulp2_id').results
+        pulp2repo1, pulp2repo2 = pulp2repositories
+        pulp3_remote1 = self.file_remotes_api.read(pulp2repo1.pulp3_remote_href)
+        pulp3_remote2 = self.file_remotes_api.read(pulp2repo2.pulp3_remote_href)
+        remote_url_repo1 = 'https://repos.fedorapeople.org/pulp/pulp/fixtures/file-many' \
+                           '/PULP_MANIFEST'
+        remote_url_repo2 = 'https://repos.fedorapeople.org/pulp/pulp/fixtures/file/PULP_MANIFEST'
+
+        self.assertEqual(pulp2repo1.pulp2_repo_id, 'file')
+        self.assertEqual(pulp2repo2.pulp2_repo_id, 'file-many')
+        self.assertTrue(pulp2repo1.is_migrated)
+        self.assertTrue(pulp2repo2.is_migrated)
+        self.assertEqual(pulp3_remote1.url, remote_url_repo1)
+        self.assertEqual(pulp3_remote2.url, remote_url_repo2)
+        self.assertEqual(pulp3_remote1.policy, 'on_demand')
+        self.assertEqual(pulp3_remote2.policy, 'immediate')
+
+    def test_distributor_different_repo(self):
+        """
+        Test that a distributor can be migrated with any Pulp 2 repo, not its native one.
+
+        Distributors are swapped in the plan.
+        """
+        self._load_and_run(DISTRIBUTOR_DIFF_PLAN)
+        pulp2repositories = self.pulp2repositories_api.list(ordering='pulp2_id').results
+        pulp2repo1, pulp2repo2 = pulp2repositories
+        pulp3_pub1 = self.file_publications_api.read(pulp2repo1.pulp3_publication_href)
+        pulp3_pub2 = self.file_publications_api.read(pulp2repo2.pulp3_publication_href)
+        pulp3_dist1 = self.file_distributions_api.read(pulp2repo1.pulp3_distribution_hrefs[0])
+        pulp3_dist2 = self.file_distributions_api.read(pulp2repo2.pulp3_distribution_hrefs[0])
+
+        self.assertEqual(pulp2repo1.pulp2_repo_id, 'file')
+        self.assertEqual(pulp2repo2.pulp2_repo_id, 'file-many')
+        self.assertTrue(pulp2repo1.is_migrated)
+        self.assertTrue(pulp2repo2.is_migrated)
+        self.assertEqual(pulp3_pub1.repository_version, pulp2repo1.pulp3_repository_version)
+        self.assertEqual(pulp3_pub2.repository_version, pulp2repo2.pulp3_repository_version)
+        self.assertEqual(pulp3_pub1.distributions[0], pulp3_dist1.pulp_href)
+        self.assertEqual(pulp3_pub2.distributions[0], pulp3_dist2.pulp_href)
+        self.assertEqual(pulp3_dist1.base_path, 'file-many')
+        self.assertEqual(pulp3_dist2.base_path, 'file')
+
+    def test_importer_no_repo(self):
+        """Test that an importer can be migrated without its native Pulp 2 repo."""
+        self._load_and_run(IMPORTER_NO_REPO_PLAN)
+        pulp2repository = self.pulp2repositories_api.list().results[0]
+        pulp3_remote = self.file_remotes_api.read(pulp2repository.pulp3_remote_href)
+        remote_url = 'https://repos.fedorapeople.org/pulp/pulp/fixtures/file-many/PULP_MANIFEST'
+
+        self.assertEqual(len(self.pulp2repositories_api.list().results), 1)
```

Comment:
Is there a particular reason to take the len of the list rather than using the count?

Reply:
(I mean that the API request returns the count as well as the list of results)

Reply:
no reason, just silly me, will fix it

Reply:
It doesn't actually matter, really nitpicky thing

Reply:
The funny thing is that I used count in some places, it's just for those calls it didn't cross my mind :shrug: 
Fixed int he next push ;)